content/guides/azure-todo-static-web-app.md

```diff
@@ -0,0 +1,747 @@
+    todoList.appendChild(li);
+  });
+}
+
+// Utility functions
+function showLoading() {
+  document.getElementById('todoList').innerHTML = '<div class="loading">Loading...</div>';
+}
+
+function hideLoading() {
+  document.getElementById('todoList').innerHTML = '';
+}
+
+function showError(message) {
+  const errorDiv = document.getElementById('error');
+  errorDiv.textContent = message;
+  setTimeout(() => (errorDiv.textContent = ''), 3000);
+}
+```
+
+The JavaScript file contains the core functions for managing todos, namely loading todos from the backend API, adding new todos, toggling completion status, and deleting todos. It also includes utility functions for showing loading indicators and error messages to the user.
+
+Let's break down the key functions and features of the todo app:
+
+Core Functions:
+
+1. `loadTodos()`
+
+   - Fetches existing todos from the Azure Functions API endpoint via `GET` request
+   - Handles loading states and error conditions
+   - Automatically called when the DOM loads
+   - Updates the UI with current todo items
+
+2. `addTodo()`
+
+   - Creates new todo items via `POST` request to the API
+   - Validates input to prevent empty submissions
+   - Updates local state and UI after successful creation
+   - Includes error handling with user feedback
+
+3. `toggleTodo(id)`
+
+   - Updates todo completion status via `PUT` request
+   - Includes error handling and state updates
+
+4. `deleteTodo(id)`
+   - Removes todos via `DELETE` request to the API
+   - Updates local state after successful deletion
+
+UI Management:
+
+1. `renderTodos()`
+   - Dynamically generates DOM elements for each todo
+   - Handles todo item styling based on completion status
+   - Creates interactive elements (checkboxes, delete buttons)
+
+Utility Functions:
+
+1. `showLoading()`
+
+   - Displays loading indicator during API operations
+   - Provides visual feedback for better user experience
+
+2. `hideLoading()`
+
+   - Removes loading indicator after operations complete
+   - Prepares UI for content display
+
+3. `showError(message)`
+   - Displays error messages to users
+   - Implements auto-dismissing notifications (3-second timeout)
+   - Provides clear feedback for error conditions
+
+State Management:
+
+- All todo items are stored in the `todos` array
+
+Error Handling:
+
+- Comprehensive try-catch blocks around API operations
+- Detailed error messages for debugging
+
+### Testing the Frontend locally
+
+To test the frontend, in your terminal, run the following command to start a local server:
+
+```bash
+swa start
+```
+
+This will start a local server on [`http://localhost:4280`](http://localhost:4280) where you can access the todo app. Open the URL in your browser to be greeted with the initial todo list interface:
+
+![Todo App Initial State](/docs/guides/swa-todo-app-intial.png)
+
+\*Note that full functionality will be available once we set up the Azure Functions backend API.
```

Comment:
is it a reference?

Reply:
I have added it just like a small note, but didn't really want to add a bigger Admonition tag, but can definitely change into that